pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for computing the pulse generator
+parameter-shift gradient of pulse sequences in a qubit-based quantum tape.
+"""
+from functools import partial
+import numpy as np
+
+import pennylane as qml
+
+from pennylane.pulse import ParametrizedEvolution
+from pennylane.ops.qubit.special_unitary import pauli_basis_strings, _pauli_decompose
+
+try:
+    import jax
+except ImportError:
+    # Handling the case where JAX is not installed is done via _assert_has_jax
+    pass
+
+
+def _one_parameter_generators(op):
+    r"""Compute the effective generators :math:`\{\Omega_k\}` of one-parameter groups that
+    reproduce the partial derivatives of a parameterized evolution.
+    In particular, compute :math:`U` and :math:`\partial U / \partial \theta_k`
+    and recombine them into :math:`\Omega_k = U^\dagger \partial U / \partial \theta_k`
+
+    Args:
+        op (`~.ParametrizedEvolution`): Parametrized evolution for which to compute the generator
+
+    Returns:
+        tuple[tensor_like]: The generators for one-parameter groups that reproduce the
+        partial derivatives of the parametrized evolution.
+        The ``k``\ th entry of the returned ``tuple`` has the shape ``(2**N, 2**N, *par_shape)``
+        where ``N`` is the number of qubits the evolution acts on and ``par_shape`` is the
+        shape of the ``k``\ th parameter of the evolution.
```

Comment:
You mean the fact that the differing part of the shapes is in the first axes? Yeah, I'm not terribly happy with it. For `SpecialUnitary` this was not an issue, and there we have the non-matrix axis in the leading position, so I wanted to be consistent.
Changing it in `SpecialUnitary` would require a deprecation cycle :grimacing: 

Regarding the proposed change: I think `parameter` is the better word in this case, because `qml.evolve(H)`  takes one argument, containing all the parameter. Does this make sense?